foyer/general_forcefield.py

```diff
@@ -0,0 +1,539 @@
+        if forcefield_files:
+            self.ff = gmso.ForceField(forcefield_files, **kwargs)
+            self._version = self.ff.version
+            self._name = self.ff.name
+            for name, atype in self.ff.atom_types.items():
+                self.atomTypeDefinitions[name] = atype.definition
+                self.atomTypeOverrides[name] = atype.overrides
+                self.atomTypeDesc[name] = atype.description
+                self.atomTypeRefs[name] = {atype.doi}
+                self.atomTypeClasses[name] = atype.atomclass
+                if atype.tags.get('element'):
+                    ele = atype.tags['element']
+                    if element.element_by_symbol(ele):
+                        self.atomTypeElements[atype.name] = ele
+                    else:
+                        self.non_element_types[ele] = None
+                else:
+                    # Register atomtype with missing atomtype as atomistic (empty string)
+                    self.atomTypeElements[name] = ''
+        return None
+
+    def apply(self, top, references_file=None, use_residue_map=True,
+              assert_bond_params=True, assert_angle_params=True,
+              assert_dihedral_params=True, assert_improper_params=True,
+              combining_rule='geometric', verbose=False, backend='gmso',
+              *args, **kwargs):
+        """Apply the force field to a molecular topology
+
+        Parameters
+        ----------
+        top : gmso.Topology, or mb.Compound
+            Molecular Topology to apply the force field to
+        references_file : str, optional, defaut=None
+            Name of file where force field refrences will be written
+            (in Bibtex format).
+        use_residue_map : bool, optional, default=True
+            Options to speed up if there are a lot of repeated
+            subtopology within the topology (assuming they all
+            have the same name).
+        assert_bond_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system bonds.
+        assert_angle_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system angles.
+        assert_dihedral_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system dihedrals.
+        assert_improper_params : bool, optional, default=True
+            If True, Foyer will exit if parametes are not found for all
+            system impropers.
+        combining_rule : str, optional, default='geometric'
+            The combining rule of the system, stored as an attribute of the
+            GMSO topology.
+        verbose : bool, optional, default=False
+            If True, Foyer will print debug-level information about notable or
+            potential problematic details it encounters.
+        backend : str, optional, default='gmso'
```

Comment:
The default backend value/string is listed. Maybe show the other options in the dockstring